File: src/runtime/nim/std/math_native.nim
# Native implementation of pytra.std.math for Nim.
#
# Provides Python math module API delegating to Nim's std/math.
# Function names match the Python originals exactly (§5.1).
#
# source: src/pytra/std/math.py

from std/math import nil

let pi*: float = math.PI
let e*: float = math.E

proc sqrt*(x: float): float = math.sqrt(x)
proc sin*(x: float): float = math.sin(x)
proc cos*(x: float): float = math.cos(x)
proc tan*(x: float): float = math.tan(x)
proc exp*(x: float): float = math.exp(x)
proc log*(x: float): float = math.ln(x)
proc log10*(x: float): float = math.log10(x)
proc fabs*(x: float): float =
  if x < 0.0: -x else: x
proc floor*(x: float): float = math.floor(x)
proc ceil*(x: float): float = math.ceil(x)
proc pow*(x: float, y: float): float = math.pow(x, y)

File: src/runtime/php/std/math_native.php
<?php
declare(strict_types=1);
/**
 * Native implementation of pytra.std.math for PHP.
 *
 * This file is copied over the linker-generated math/east.php at emit time.
 * The generated module body references the Python `math` module which is
 * invalid in PHP; this native seam delegates to PHP built-in functions.
 *
 * Function/variable names match the EAST module exports so that callers
 * resolved by the linker (`sqrt($x)`, `pi`, etc.) work directly.
 *
 * source: src/pytra/std/math.py
 */

if (!defined('pi')) { define('pi', M_PI); }
if (!defined('e'))  { define('e',  M_E);  }

// PHP built-in: sqrt, sin, cos, tan, exp, log, log10, floor, ceil, pow
// are already available as global functions.

// fabs is not a PHP built-in; delegate to abs().
if (!function_exists('fabs')) {
    function fabs($x) { return abs((float)$x); }
}
